src/libexpr/primops.cc

```diff
@@ -1664,9 +1664,56 @@ static void prim_findFile(EvalState & state, const PosIdx pos, Value * * args, V
     v.mkPath(state.checkSourcePath(state.findFile(searchPath, path, pos)));
 }
 
-static RegisterPrimOp primop_findFile(RegisterPrimOp::Info {
+static RegisterPrimOp primop_findFile(PrimOp {
     .name = "__findFile",
-    .arity = 2,
+    .args = {"search path", "lookup path"},
+    .doc = R"(
+      Look up the given path with the given search path.
+
+      A search path is represented list of [attribute sets](./values.md#attribute-set) with two attributes, `prefix`, and `path`.
+      `prefix` is an relative path.
+      `path` denotes a directory; exactly what sort of value it is is however is an implementation detail.
```

Comment:
It currently is just what the user wrote on the command line: URL (`flake:...`, `chanel:...`, etc.) / path / etc.

I don't like how that stuff leaks into the language, so I rather not document it.

(And In the future, I think we could replace with an opaque fetcher object that one cannot inspect either. Indeed. this might be a good first use of lazy trees.)

Reply:
@fricklerhandwerk Feel free to take a machete to this section. Really no one should be using `builtins.findFile` yet I think, it is too crufty at leaking internals at this point. I don't want to make things seem more stable by the mere fact of documenting how they work.

Reply:
```suggestion
      `path` denotes a location; the exact syntax depends on the command line interface.
```

Could be a file I think. Referring to the CLI seems like a sensible cop-out for now, so that we don't have to over-document this wart.

Reply:
```suggestion
      `path` denotes a file system location; the exact syntax depends on the command line interface.
```

The CLI bit is not exactly true though, since we also have the `nix-path` setting which we should link to and use as the single source of truth for documenting the outward interface (which it's currently not).